hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java

```diff
@@ -182,8 +183,21 @@ public void newDead(ServerName sn) {
   }
 
   private void spawnRenewalChore(final UserGroupInformation user) {
-    authService = new ChoreService("Relogin service");
-    authService.scheduleChore(AuthUtil.getAuthRenewalChore(user));
+    ChoreService service = getChoreService();
+    service.scheduleChore(AuthUtil.getAuthRenewalChore(user));
+  }
+
+  /**
+   * If choreService has not been created yet, create the ChoreService.
+   * @return ChoreService
+   */
+  ChoreService getChoreService() {
+    synchronized (this) {
+      if (choreService == null) {
+        choreService = new ChoreService("AsyncConn Chore Service");
+      }
+    }
+    return choreService;
```

Comment:
Move the return into the synchronized block? Or if you want to use DCL, declare choreService as volatile and test it first before the synchronized.

Reply:
Done  with the first approach.